prow/kube/client.go

```diff
@@ -104,8 +104,10 @@ func (c *Client) requestRetry(r *request) ([]byte, error) {
 	backoff := retryDelay
 	for retries := 0; retries < maxRetries; retries++ {
 		resp, err = c.doRequest(r.method, r.path, r.query, r.requestBody)
-		if err == nil {
+		if err == nil && resp.StatusCode < 500 {
```

Comment:
maybe nest the resp.StatusCode < 500 insidE?

    if err == nil {
        if resp.StatusCode < 500 {
            break
        }
        resp.Body.Close()
    }